pallets/staking/src/lib.rs

```diff
@@ -2156,14 +2208,25 @@ decl_module! {
         /// # Arguments
         /// * origin Required origin for adding a potential validator.
         /// * identity Validator's IdentityId.
+        /// * intended_count No. of validators given identity intended to run.
         #[weight = 750_000_000]
-        pub fn add_permissioned_validator(origin, identity: IdentityId) {
+        pub fn add_permissioned_validator(origin, identity: IdentityId, intended_count: Option<u32>) {
             T::RequiredAddOrigin::ensure_origin(origin)?;
-            ensure!(!Self::permissioned_identity(&identity), Error::<T>::AlreadyExists);
+            ensure!(Self::permissioned_identity(&identity).is_none(), Error::<T>::AlreadyExists);
             // Validate the cdd status of the identity.
             ensure!(<Identity<T>>::has_valid_cdd(identity), Error::<T>::InvalidValidatorIdentity);
+            let pref = match intended_count {
+                Some(count) => {
+                    // Maximum allowed validator count is always less than the 2/3 of validator_count().
+                    let allowed_validator_count = T::MaxValidatorPerIdentity::get() * Self::validator_count();
```

Comment:
Let's extract this computation into a function since it exists elsewhere in the pallet.